secure/net/src/network_controller/mod.rs

```diff
@@ -144,6 +193,10 @@ mod tests {
         network_controller1.start();
         network_controller2.start();
 
+        // wait for the server to be ready to serve
+        // TODO: We need to pass this test without this sleep
+        thread::sleep(std::time::Duration::from_millis(10));
```

Comment:
See my comment on using a one shot channel to ensure the server is started.. we can probably get rid of the sleep here..

Reply:
yes one shot channel will help here

Reply:
one shot channel will not help as explained above. Closing this comment as duplicate